docs/classes.md

```diff
@@ -0,0 +1,263 @@
+# Supported Class by Pyccel
+
+Pyccel strives to provide robust support for object-oriented programming concepts commonly used by developers. In Pyccel, classes are a fundamental building block for creating structured and reusable code. This documentation outlines key features and considerations when working with classes in Pyccel.
+
+## Constructor Method
+
+-   The Constructor Method, `__init__`, is used to initialize the object's attributes.
+-   It must be named `__init__`, or the Pyccel compiler will generate an error.
+-   The first parameter of the `Constructor Method` should always be named `self`.
+-   Pyccel doesn't support Interface Constructor.
+
+## Destructor Method
+
+-   The Destructor Method, `__del__`, is used to perform cleanup actions when an object is destroyed.
+-   Pyccel automatically adds attributes that need to be freed to the garbage collector. These attributes are subsequently released during the destructor's execution. This mechanism helps manage memory efficiently, ensuring that resources are appropriately deallocated.
+-   One notable feature of Pyccel is its automatic management of destructor calls, even when constructors are invoked in different scopes. This ensures that resources are properly cleaned up when class instances are no longer in use.
+-   The first parameter of the `Destructor Method` should always be named `self`.
```

Comment:
Isn't this true for any class method not just for `__init__` and `__del__`?

Reply:
I've just seen this is mentioned below. I don't know if it's worth repeating 3 times. Especially as you are already working on removing this restriction (#1479) so you will need to delete this in 3 places. If you don't mind deleting it 3 times though then why not.

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1580/commits/f92c3e1883b231711e3cbad1f475439610d22f96.